src/util/include/util/serializing/ObjectInputStream.h

```diff
@@ -39,17 +41,51 @@ class ObjectInputStream {
     void readData(std::vector<T>& data);
 
     /// Reads raw image data from the stream.
-    std::string readImage();
+    std::vector<std::byte> readImage();
 
 private:
-    void checkType(char type);
+    void checkType(const std::byte type);
 
-    static std::string getType(char type);
+    static std::string getType(const std::byte type);
 
 private:
     std::istringstream istream;
```

Comment:
Wouldn't a std::basic_istream<std::byte> make more sense then?

Reply:
I also wondered if that would be a good alternative to char streams, but after reading https://open-std.org/JTC1/SC22/WG21/docs/papers/2020/p2146r2.pdf , I think this won't work, because std::byte is not a char type and therefore does not implement char_traits. I think, we should stay with the old streams, until a good alternative is accepted.